### DISCUSSION TITLE
[Feature Request] Dynamic Layout

### DISCUSSION THREAD

**Discussion by jalalmanafi** [heroui-inc/heroui#4730] [Feature requests]
### Is your feature request related to a problem? Please describe.

It would be perfect to write dynamic layout like this [repository](https://github.com/bvaughn/react-resizable-panels)

### Describe the solution you'd like

[DEMO](https://react-resizable-panels.vercel.app/examples/horizontal)

### Describe alternatives you've considered

For now I do not have any alternative variant

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1419/[feature-request]-dynamic-layout">ENG-1419 [Feature Request] Dynamic Layout</a></p>
